### PR TITLE
Create Header: Ruby generated create options

### PR DESCRIPTION
**Pull request by Erin007**
Original PR: #24994 reverted in #25006 due to https://codedotorg.slack.com/archives/C0T0PNTM3/p1537892800000100. 

This PR includes the original changes, plus `require 'cdo/create_header'` in `dashboard/app/views/layouts/_header.html.haml` which should fix the error the led to the reversion.

### CODE REVIEW

File: lib/cdo/create_header.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# dropdown in the header.
+
+class CreateHeader
+  def self.get_create_dropdown_contents(options)
+    everyone_entries = [
+      {
+        id: "create_dropdown_playlab",
+        title: I18n.t("nav.create_button.playlab"),
+        url: CDO.studio_url('projects/playlab/new'),
+        image: "/shared/images/fill-70x70/courses/logo_playlab.png"
```
Comment (breville) [inline lib/cdo/create_header.rb:11]: idea: you could factor out the common part of the image URL and put it in the code at the end of this function.

File: lib/cdo/create_header.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        url: CDO.studio_url('projects/gamelab/new'),
+        image: "/shared/images/fill-70x70/courses/logo_gamelab_square.png"
+      },
+    ]
+
+    entries = options[:user_under_13] ?
+      everyone_entries + younger_than_13_entries :
+      everyone_entries + older_than_13_entries
+
+    entries
```
Comment (breville) [inline lib/cdo/create_header.rb:55]: nit: don't need the `entries` variable at all, though maybe this does read more clearly.

File: shared/haml/user_header.haml [hunk 3]
```diff
@@ -9,32 +14,11 @@
...
-        %a.project_link_box{href: CDO.studio_url('projects/gamelab/new')}
-          %img{src: "/shared/images/fill-70x70/courses/logo_gamelab_square.png"}
-          .project_link
-            = I18n.t("nav.create_button.gamelab")
+      - create_drop_down_items.each do |entry|
+        %a.project_link_box{href: entry[:url]}
+          %img{src: entry[:image]}
+          .project_link{ id: entry[:id]}
+            .text= entry[:title]
       %a.project_link_box{href: CDO.studio_url('projects/public')}
```
Comment (breville) [inline shared/haml/user_header.haml:22]: Love how much HAML we get to chop here.

File: lib/cdo/create_header.rb (hunk 1 above)
Comment (Erin007) [reply lib/cdo/create_header.rb:11]: Oooh. I like that idea. I'll incorporate in my next PR.

**Pull request merged**